### ISSUE TITLE
Crash loop for kubernetes-ingress after upgrading to 1.7

### ISSUE THREAD

**Issue by ShreyShah977** [haproxytech/kubernetes-ingress#414]
### Intro

Hello! My team and I are running a cloud deployment on GCR, and leveraging the HAProxy ingress controller in our setup.
We are running the `kubernetes-ingress` via leveraging the helm chart.

### Some notes of interest of our configuration:
1. We are running it in high availability mode.
2. Enabling pod security (i.e `podSecurityPolicy.enabled` is set to `true`)

### Errors currently experienced when the pod for ingress controller is launched.

1. 
```
2022/01/14 18:33:58 ERROR   rules/main.go:199 failed to create a REQ_INSPECT_DELAY rule: 14: unsupported action in tcp_request_rule
2022-01-14 13:33:58	
(no unique labels)
2022/01/14 18:33:58 ERROR   rules/main.go:199 failed to create a REQ_INSPECT_DELAY rule: 14: unsupported action in tcp_request_rule
```

And

2. 
```
E0114 18:33:58.916687     211 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167: 
Failed to watch *v1alpha1.Global: failed to list *v1alpha1.Global: globals.core.haproxy.org is forbidden: 
User "system:serviceaccount:ingress-ctrl:ingress-ctrl-kubernetes-ingress" cannot list resource "globals" in API group "core.haproxy.org" at the cluster scope
```

### Expected Results

We have used HAProxy previously (`img` version 1.6.7). However with the new update to 1.7.4 and the chart being updated to 1.18.1, we are consistently encountering the ingress controller pods restarting every time those above errors are shown in the logs. It performs startup procedure, showcases the error messages and then promptly shuts down and repeats.

Additionally we have the kubernetes-ingress chart as a subchart, with our `values.yaml` file below overriding it.

Also attaching the config file: `values.yaml` to show how we configure it with our system.

Thanks and appreciate any and all support.
[values.yaml.txt](https://github.com/haproxytech/kubernetes-ingress/files/7885082/values.yaml.txt)

**Comment by uxes** [comment]
I'v encountered same issue, using v1.6.9 has helped for me (anyway i had to set statically version in DaemonSet)

**Comment by Mo3m3n** [comment]
It would be great if anyone can provide/isolate the config that triggers this bug.
I am suspecting this is happening with ingress resources annotated with ssl-passthrough, but I was not able to reproduce with a typical ssl-passthrough config.

**Comment by cristian-aldea** [comment]
> I'v encountered same issue, using v1.6.9 has helped for me (anyway i had to set statically version in DaemonSet)

@uxes Do you have an ingress resource with ssl-passthrough? Just double-checking if this might be indeed the common link

**Comment by uxes** [comment]
> > I'v encountered same issue, using v1.6.9 has helped for me (anyway i had to set statically version in DaemonSet)
> 
> @uxes Do you have an ingress resource with ssl-passthrough? Just double-checking if this might be indeed the common link

@cristian-aldea  Sure, I use HAProxy with SSL Pass-Through (inspired by those articles https://serversforhackers.com/c/using-ssl-certificates-with-haproxy https://akomljen.com/get-automatic-https-with-lets-encrypt-and-kubernetes-ingress/).

**Comment by cristian-aldea** [comment]
I see. Just letting you know that it seems that the issue is related to the ssl-passthrough configuration. Once we removed the ingress which had that configuration the latest release started to work for us. Something to keep in mind.

**Comment by Mo3m3n** [comment]
I was able to reproduce this issue.
The problem is about  using [ssl-passthrough](https://github.com/haproxytech/kubernetes-ingress/tree/master/documentation#ssl-passthrough) without setting a [client-timeout](https://github.com/haproxytech/kubernetes-ingress/tree/master/documentation#timeout-client).  
Reason is that when ssl-passthrough is used controller inserts a [tcp-request inspect-delay](https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#4-tcp-request%20inspect-delay) this makes haproxy wait for data in order to correctly inspect the content and take decisions. 
The delay value is equal to [client-timeout](https://github.com/haproxytech/kubernetes-ingress/tree/master/documentation#timeout-client) if it is set otherwise controller falls back to a hardcoded value (5s). The regression here is with the fallback mechanism that is not working in 1.7 that is why the bug is triggered only if there is no timeout set.
So the quick fix is to configure client-timeout (in configmap) this is a good practice anyway as we don't want connections to stay open indefinitely without sending data.
Patch is on the way.
Thank you for the input which helped narrow down this bug.

**Comment by ShreyShah977** [comment]
Update. Added the `ingress.kubernetes.io/timeout-client: '5000'` into the `annotations` for our ingress resource.
It is now working as expected. Thank you for your efforts on this issue.

**Comment by Mo3m3n** [comment]
Fixed in https://github.com/haproxytech/kubernetes-ingress/commit/1966e37d7315be071fe15c4321fce2f9aaf5578e
I will update this thread when next version is out.

**Comment by Siegfriedk** [comment]
Just being curious: @Mo3m3n are there no tests in the haproxy ingress? Or is that git commit not everything?

**Comment by Siegfriedk** [comment]
I meant your git commit you posted above me. It did not reference a PR or something :)

**Comment by Mo3m3n** [comment]
Ah right, currently we work with an internal CI/CID pipelines (gitlab) where internal repo is synced with this one.
PRs are welcome though

**Comment by Mo3m3n** [comment]
Fix available in `v1.7.5`
